core/manager.go

```diff
@@ -0,0 +1,103 @@
+// Copyright 2019 ChainSafe Systems (ON) Corp.
+// This file is part of gossamer.
+//
+// The gossamer library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The gossamer library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the gossamer library. If not, see <http://www.gnu.org/licenses/>.
+
+package core
+
+import (
+	log "github.com/ChainSafe/log15"
+
+	scale "github.com/ChainSafe/gossamer/codec"
+	tx "github.com/ChainSafe/gossamer/common/transaction"
+	"github.com/ChainSafe/gossamer/consensus/babe"
+	"github.com/ChainSafe/gossamer/core/types"
+	"github.com/ChainSafe/gossamer/p2p"
+	"github.com/ChainSafe/gossamer/runtime"
+)
+
+type Service struct {
+	rt *runtime.Runtime
+	b  *babe.Session
+
+	msgChan <-chan p2p.Message
```

Comment:
I don't think there is any way to get this and write to it (eg. to pass into the p2p service)

Reply:
it's not passed into the p2p service, it's created in the p2p service then we get it from there